### DISCUSSION TITLE
Roadmap for preview feature admin-fine-grained-authz

### DISCUSSION THREAD

**Discussion by jowko** [keycloak/keycloak#25489] [Q&A]
Hello.

There are a few preview features which are listed here: https://www.keycloak.org/server/features#_preview_features
I tried to find any info about delivery date for **admin-fine-grained-authz** but I did not found it. Does keycloak team plan to move this out of preview in 2024?

I also tried to find roadmap for 2023/2024 but I did not found it. There is just discussion about it: https://github.com/keycloak/keycloak/discussions/17096

I wanted to use that feature to enable impersonation for certain users and limit who can be impersonated. This feature fulfils this requirement but we cannot go to production with it because it is in preview state.

We are also interested in token-exchange but this is also a preview feature. Any plans on that?

**Comment by bernhof** [comment]
Not an answer but just additional info from back in 2020: https://groups.google.com/g/keycloak-dev/c/CgF8ZuNqVRg/m/ShX8H7ZHBAAJ

> As it is a preview feature there is no guarantee that it won't be removed or drastically changed. The current plans is to review and improve what we have today, but I'm afraid it's not our highest priority right now so it may happen during 2020, or it may be postponed to 2021.

We, too, would appreciate an updated status on this feature, since it has potential to save us a lot of custom development
